GHzDACs/ghz_fpga_server.py

```diff
@@ -1631,16 +1655,100 @@ def dac_run_sram(self, c, data, loop=False, blockDelay=0):
         """
         if not len(data):
             return
-        
-        if loop:
-            # make sure data is at least 20 words long by repeating it
-            data *= (20-1)/len(data) + 1
-        else:
-            # Set data at least 20 words long by repeating first value
+
+        if len(data) < 20:
+            # pad out to 20 with zeros
```

Comment:
I'm going to change this to throw an exception if you try to give less than 20 ns of data. We don't do so anywhere (I'm fairly sure--I'm at least familiar with the cals and the bringup), and it's better not to mutate the data that the user passes in.